src/core/Akka.Streams/Implementation/ActorRefSourceActor.cs

```
@@ -63,13 +63,13 @@ public ActorRefSourceActor(int bufferSize, OverflowStrategy overflowStrategy, in
         {
             BufferSize = bufferSize;
             OverflowStrategy = overflowStrategy;
-            Buffer = bufferSize != 0 ? Implementation.Buffer.Create<T>(bufferSize, maxFixedBufferSize) : null;
+            Buffer = bufferSize > 0 ? Implementation.Buffer.Create<T>(bufferSize, maxFixedBufferSize) : null;
```

LGTM

---

Although, should we throw an exception here if the `bufferSize` is less than `0` ?

---

I think we should, I just tried to preserve the old behavior